### PR TITLE
Port debugger tests to new test harness

### PR DESCRIPTION
**Pull request by sduquej**
For https://github.com/glimmerjs/glimmer-vm/issues/561

### CODE REVIEW

**Comment by sduquej** [comment]
@chadhietala would appreciate :eyes: on this. Specially around how to best DRY things up with the new harness.

File: packages/@glimmer/runtime/test/debugger-test.ts [hunk 1]
```diff
@@ -1,82 +1,94 @@
...
-  self = new UpdatableReference(context);
-  env.begin();
-  let templateIterator = template.render({ self, parentNode: root, dynamicScope: new TestDynamicScope() });
-  let iteratorResult: IteratorResult<RenderResult>;
-  do {
-    iteratorResult = templateIterator.next();
-  } while (!iteratorResult.done);
+    this.assertStableRerender();
+    this.assert.equal(callbackExecuted, 1);
+    this.assertHTML('true');
```
Comment (chadhietala) [inline packages/@glimmer/runtime/test/debugger-test.ts]: You can ✂️ this line as L24 is checking the same thing.

File: packages/@glimmer/runtime/test/debugger-test.ts [hunk 2]
```diff
@@ -1,82 +1,94 @@
...
-}
+    expectedContext = {
+      foo: 'baz',
+      a: {
+        b: false
+      }
+    };
+    this.rerender(expectedContext);
+    this.assert.equal(callbackExecuted, 2);
+    this.assertHTML('false');
```
Comment (chadhietala) [inline packages/@glimmer/runtime/test/debugger-test.ts]: Can you add `this.assertStableNodes()` on the next line

File: packages/@glimmer/runtime/test/debugger-test.ts [hunk 3]
```diff
@@ -1,82 +1,94 @@
...
-    resetDebuggerCallback();
+    expectedContext = {
+      foo: 'bar',
+      a: {
+        b: true
+      }
+    };
+    this.rerender(expectedContext);
+    this.assert.equal(callbackExecuted, 3);
+    this.assertHTML('true');
```
Comment (chadhietala) [inline packages/@glimmer/runtime/test/debugger-test.ts]: Same comment about adding `this.assertStableNodes()`

File: packages/@glimmer/runtime/test/debugger-test.ts [hunk 4]
```diff
@@ -1,82 +1,94 @@
...
-function render<T>(template: Template<T>, context={}) {
-  self = new UpdatableReference(context);
-  env.begin();
-  let templateIterator = template.render({ self, parentNode: root, dynamicScope: new TestDynamicScope() });
-  let iteratorResult: IteratorResult<RenderResult>;
-  do {
-    iteratorResult = templateIterator.next();
-  } while (!iteratorResult.done);
+    this.assertStableRerender();
+    this.assert.equal(callbackExecuted, 1);
```
Comment (chadhietala) [inline packages/@glimmer/runtime/test/debugger-test.ts]: No need for this guy ✂️

File: packages/@glimmer/runtime/test/debugger-test.ts [hunk 5]
```diff
@@ -1,82 +1,94 @@
...
-  });
-});
+    this.render('{{#with foo as |bar|}}{{#if a.b}}true{{debugger}}{{else}}false{{debugger}}{{/if}}{{/with}}', expectedContext);
+    this.assert.equal(callbackExecuted, 1);
+    this.assertHTML('true');
 
-QUnit.test('can get locals', assert => {
-  let template = compile(`{{#with foo as |bar|}}{{debugger}}{{/with}}`);
+    this.assertStableRerender();
+    this.assert.equal(callbackExecuted, 1);
```
Comment (chadhietala) [inline packages/@glimmer/runtime/test/debugger-test.ts]: ✂️ 70 & 71

File: packages/@glimmer/runtime/test/debugger-test.ts [hunk 6]
```diff
@@ -1,82 +1,94 @@
...
-  });
+    expectedContext = {
+      foo: 'baz',
+      a: {
+        b: false
+      }
+    };
+    this.rerender(expectedContext);
+    this.assert.equal(callbackExecuted, 2);
+    this.assertHTML('false');
```
Comment (chadhietala) [inline packages/@glimmer/runtime/test/debugger-test.ts]: Add `this.assertStableNodes()`

File: packages/@glimmer/runtime/test/debugger-test.ts [hunk 7]
```diff
@@ -1,82 +1,94 @@
...
-});
+    expectedContext = {
+      foo: 'bar',
+      a: {
+        b: true
+      }
+    };
+    this.rerender(expectedContext);
+    this.assert.equal(callbackExecuted, 3);
+    this.assertHTML('true');
```
Comment (chadhietala) [inline packages/@glimmer/runtime/test/debugger-test.ts]: Add `this.assertStableNodes()`

**Comment by sduquej** [comment]
@chadhietala Done! Thanks for the quick review ⚡️

**Comment by GavinJoyce** [comment]
@sduquej there seems to be a couple of failing tests:

```
not ok 270 PhantomJS 2.0 - [NEW] Debugger Tests: basic debugger statement
    ---
        actual: >
            Max depth.,Max depth.
        expected: >
            Max depth.,Max depth.
        stack: >
            	at assertStableNodes (http://localhost:7000/2694/tests/assets/glimmer.js:14303:34)
            	at http://localhost:7000/2694/tests/assets/tests.js:3033:35
            	at http://localhost:7000/2694/tests/assets/glimmer.js:14523:43
            	at runTest (http://localhost:7000/2694/tests/assets/qunit.js:1286:34)
            	at run (http://localhost:7000/2694/tests/assets/qunit.js:1272:13)
            	at http://localhost:7000/2694/tests/assets/qunit.js:1450:15
            	at process (http://localhost:7000/2694/tests/assets/qunit.js:2684:26)
            	at begin (http://localhost:7000/2694/tests/assets/qunit.js:2667:11)
            	at http://localhost:7000/2694/tests/assets/qunit.js:2627:11
        message: >
            DOM nodes are stable
        Log: |
    ...
not ok 271 PhantomJS 2.0 - [NEW] Debugger Tests: can get locals
    ---
        actual: >
            Max depth.,Max depth.
        expected: >
            Max depth.,Max depth.
        stack: >
            	at assertStableNodes (http://localhost:7000/2694/tests/assets/glimmer.js:14303:34)
            	at http://localhost:7000/2694/tests/assets/tests.js:3073:35
            	at http://localhost:7000/2694/tests/assets/glimmer.js:14523:43
            	at runTest (http://localhost:7000/2694/tests/assets/qunit.js:1286:34)
            	at run (http://localhost:7000/2694/tests/assets/qunit.js:1272:13)
            	at http://localhost:7000/2694/tests/assets/qunit.js:1450:15
            	at process (http://localhost:7000/2694/tests/assets/qunit.js:2684:26)
            	at begin (http://localhost:7000/2694/tests/assets/qunit.js:2667:11)
            	at http://localhost:7000/2694/tests/assets/qunit.js:2627:11
        message: >
            DOM nodes are stable
        Log: |
    ...
```

File: packages/@glimmer/runtime/test/debugger-test.ts (hunk 2 above)
Comment (sduquej) [inline packages/@glimmer/runtime/test/debugger-test.ts]: Had failing specs as `this.snapshot` is outdated in `assertStableNodes()` after the rerender
https://github.com/glimmerjs/glimmer-vm/blob/aa0ddc6ee44083a14aa690f7f212e7f67127ab1c/packages/%40glimmer/test-helpers/lib/abstract-test-case.ts#L522-L526

I resorted to calling `assertStableRerender` to have a new snapshot taken but feel that it isn't the right thing to do. Would appreciate some input regarding this @chadhietala 🙇

**Comment by chadhietala** [comment]
Thanks @sduquej! I merged these as part of #627.

**Pull request closed without merging**